### PR TITLE
Update aeson to greater than 2

### PR DESCRIPTION
**Pull request by 0xr0bert**
aeson >=2 Introduced the Data.Aeson.Key.Key type and replaced any Text keys with Key.

Builds & tests ok

### CODE REVIEW

**Comment by 0xr0bert** [comment]
Closes #9 

If ok would you release to hackage please!

File: src/Network/API/TheMovieDB/Internal/Settings.hs [hunk 1]
```diff
@@ -24,7 +24,7 @@ import Network.API.TheMovieDB.Internal.Types
 -- | Settings used by this library.
 data Settings = Settings
   { -- | The API key to use.
-    tmdbKey :: Key,
+    tmdbKey :: Network.API.TheMovieDB.Internal.Types.Key,
```
Comment (pjones) [inline src/Network/API/TheMovieDB/Internal/Settings.hs:27]: @ragreener1 Can you explain why this change was needed.  Thank you.

File: src/Network/API/TheMovieDB/Internal/Settings.hs [hunk 2]
```diff
@@ -43,7 +43,7 @@ instance ToJSON Settings where
       ]
 
 -- | Default settings.
-defaultSettings :: Key -> Settings
+defaultSettings :: Network.API.TheMovieDB.Internal.Types.Key -> Settings
```
Comment (pjones) [inline src/Network/API/TheMovieDB/Internal/Settings.hs:46]: Same as above.

**Comment by pjones** [comment]
I will definitely get this out to hackage as soon as this is merged.

**Review by pjones — CHANGES_REQUESTED** [review]

File: src/Network/API/TheMovieDB/Internal/Settings.hs (hunk 1 above)
Comment (0xr0bert) [reply src/Network/API/TheMovieDB/Internal/Settings.hs:27]: Sure. The new version of Aesop introduces a Key type. This clashes with your key type, so we need to be explicit

**Comment by pjones** [comment]
@ragreener1 Can you enable "allow maintainers to add commits" so I can clean up imports and them merge?  Thank you.

**Pull request merged**